### PR TITLE
Make phone number required if preferred contact method

### LINKED ISSUE #3159: Conditional logic for phone number
Trigger error if phone is selected as preferred contact method but no phone number is listed.

### PR DESCRIPTION
**Pull request by alexose**
Fixes #3159

### CODE REVIEW

File: src/js/common/components/questions/Phone.jsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ class Phone extends React.Component {
             autocomplete="tel"
             placeholder="xxxxxxxxxx"
             field={this.props.value}
+            required={this.props.required}
```
Comment (akainic) [inline src/js/common/components/questions/Phone.jsx]: Did we verify that this doesn't create issues for other projects using this component?

**Review by akainic — COMMENTED** [review]
I wonder if this is the best UX implementation for this (showing an error on a previous field without really explaining why), or if we need an accompanying error message to appear near the preferred contact method question, as that is the question that will trigger this error. Thoughts? Also @cweber and @goldenmeanie

File: src/js/common/components/questions/Phone.jsx (hunk 1 above)
Comment (alexose) [inline src/js/common/components/questions/Phone.jsx]: I looked at the other instance of `Phone` and didn't see anything.  And tests are still passing :)

**Comment by alexose** [comment]
We could switch the order of the fields.  I think that would be a little nicer

**Comment by cweber** [comment]
I see the confusion. I think maybe a reasonable solution would be to order the questions on that page as:

1. Preferred method of contact
2. Address
3. Email
4. Phone [clear error message about one is triggered because you selected it as your preferred method]

@goldenmeanie Do you concur?

**Comment by alexose** [comment]
Question order updated.  I also added a unit test.

As far as additional error messaging, it's a little tricky to figure out where to put it.  Is this amount of red text overwhelming?

image

**Comment by cweber** [comment]
It is quite a lot of text.

- I would put the message about 'preferred' first when it's true
- Also...can we modify how the phone number field validates so we can remove some of that warning message? Handling dashes seems like something we should be able to handle, rather than putting that on the person entering the data. (remove non numeric characters and then count?)

**Comment by alexose** [comment]
* Reversed the order of the error message
* I like this idea-- It's good for users.  We don't really do anything like this yet, so it might take some thought, e.g., do we strip the dashes on submit?  On blur?  Do we just prevent them from typing dashes to begin with?

I updated the way the validation works, but I still need to prevent us from sending dashes to the API.

**Comment by cweber** [comment]
I think ideally we'd let them type it however they want so it looks as they'd expect it and then strip out any non-numbers so we can validate that there are 9 digits.

That said, should our forms limit people to US numbers? I mean, maybe yes, because US veterans, but is this the equivelant of limiting address fields to only domestic?

@goldenmeanie Any thoughts?

**Comment by goldenmeanie** [comment]
Is there some international friendly phone validation pattern out there? If not, defaulting to US seems like a reasonable fall-back.

**Comment by akainic** [comment]
Let's leave internationalizing the phone field for another PR, I'm going to update this one and try to get it merged as is for now.

**Pull request merged**